src/trackedit/internal/au3/au3selectioncontroller.h

```diff
@@ -69,13 +69,13 @@ class Au3SelectionController : public ISelectionController
     };
 
     // track selection
-    Val<trackedit::TrackId> m_selectedTrack;
+    Val<TrackIdList> m_selectedTracks;
 
     // clip selection
     Val<trackedit::ClipKey> m_selectedClip;
 
     // data selection
-    Val<std::vector<trackedit::TrackId>> m_selectedTrackIds;
+    Val<TrackIdList> m_dataSelectedTrackIds;
```

Comment:
As discussed 1-1: there is no plan for discontinuous time selection (@LWinterberg please confirm), hence the time selection can entirely be determined by m_selectedStartTime and m_selectedEndTime.

Reply:
In Audacity 3, some logic like the generators determine where to generate based on this time selection and the (possibly discontinuous) set of selected tracks.
In Audacity 4, with discontinuous clip selection, it seems to me like there is going to be an additional, prevailing logic: generators will first check if there are selected clips (`m_selectedClip` will by then be `m_selectedClips`) and use this as placeholder, else it will fall back on the Au3 behavior.